iset.mm

```diff
@@ -125850,6 +125890,157 @@ _Introduction to General Topology_ (1983), p. 114) and it is convenient
+
+  ${
+    $d B b j $.  $d J b j $.
+    $( Property of being a topology with a given base set.  (Contributed by
+       Stefan O'Rear, 31-Jan-2015.)  (Revised by Mario Carneiro,
+       13-Aug-2015.) $)
+    istopon $p |- ( J e. ( TopOn ` B ) <-> ( J e. Top /\ B = U. J ) ) $=
+      ( vj vb ctopon cfv wcel cvv ctop cuni wceq wa cdm wrel wfun funtopon crab
+      cv cpw wss funrel ax-mp relelfvdm elexd uniexg eleq1 syl5ibrcom imp eqeq1
+      mpan rabbidv df-topon vpwex pwex wi rabss pwuni pweq syl5sseqr sylibr a1i
+      selpw mprgbir ssexi fvmpt3i eleq2d unieq eqeq2d elrab syl6bb pm5.21nii )
+      BAEFZGZAHGZBIGZABJZKZLZVMAEMZENZVMAVSGEOVTPEUAUBBAEUCUJUDVOVQVNVOVNVQVPHG
+      BIUEAVPHUFUGUHVNVMBACRZJZKZCIQZGVRVNVLWDBDADRZWBKZCIQZWDHEWEAKWFWCCIWEAWB
+      UIUKCDULWGWESZSZWHDUMUNWGWITWFWAWIGZUOZCIWFCIWIUPWKWAIGWFWAWHTWJWFWBSWAWH
+      WAUQWEWBURUSCWHVBUTVAVCVDVEVFWCVQCBIWABKWBVPAWABVGVHVIVJVK $.
+  $}
+
+  $( A topology on a given base set is a topology.  (Contributed by Mario
+     Carneiro, 13-Aug-2015.) $)
+  topontop $p |- ( J e. ( TopOn ` B ) -> J e. Top ) $=
+    ( ctopon cfv wcel ctop cuni wceq istopon simplbi ) BACDEBFEABGHABIJ $.
+
+  $( The base set of a topology on a given base set.  (Contributed by Mario
+     Carneiro, 13-Aug-2015.) $)
+  toponuni $p |- ( J e. ( TopOn ` B ) -> B = U. J ) $=
+    ( ctopon cfv wcel ctop cuni wceq istopon simprbi ) BACDEBFEABGHABIJ $.
+
+  ${
+    topontopi.1 $e |- J e. ( TopOn ` B ) $.
+    $( A topology on a given base set is a topology.  (Contributed by Mario
+       Carneiro, 13-Aug-2015.) $)
+    topontopi $p |- J e. Top $=
+      ( ctopon cfv wcel ctop topontop ax-mp ) BADEFBGFCABHI $.
+
+    $( The base set of a topology on a given base set.  (Contributed by Mario
+       Carneiro, 13-Aug-2015.) $)
+    toponunii $p |- B = U. J $=
+      ( ctopon cfv wcel cuni wceq toponuni ax-mp ) BADEFABGHCABIJ $.
+  $}
+
+  ${
+    toptopon.1 $e |- X = U. J $.
+    $( Alternative definition of ` Top ` in terms of ` TopOn ` .  (Contributed
+       by Mario Carneiro, 13-Aug-2015.) $)
+    toptopon $p |- ( J e. Top <-> J e. ( TopOn ` X ) ) $=
+      ( ctopon cfv wcel ctop cuni wceq istopon mpbiran2 bicomi ) ABDEFZAGFZMNBA
+      HICBAJKL $.
+  $}
+
+  $( A topology is the same thing as a topology on the union of its open sets.
+     (Contributed by BJ, 27-Apr-2021.) $)
+  toptopon2 $p |- ( J e. Top <-> J e. ( TopOn ` U. J ) ) $=
+    ( cuni eqid toptopon ) AABZECD $.
+
+  $( A topology on a set is a topology on the union of its open sets.
+     (Contributed by BJ, 27-Apr-2021.) $)
+  topontopon $p |- ( J e. ( TopOn ` X ) -> J e. ( TopOn ` U. J ) ) $=
+    ( ctopon cfv wcel ctop cuni topontop toptopon2 sylib ) ABCDEAFEAAGCDEBAHAIJ
+    $.
+
+  ${
+    $d A x y $.
+    $( The set of topologies on a set is included in the double power set of
+       that set.  (Contributed by BJ, 29-Apr-2021.)  (Revised by Jim Kingdon,
+       16-Jan-2023.) $)
+    toponsspwpwg $p |- ( A e. V -> ( TopOn ` A ) C_ ~P ~P A ) $=
```

Comment:
in both set/iset, the conclusion can be strengthened to ``( TopOn ` A ) e. ~P ~P ~P A ``. Is it useful ?

Reply:
Since `toponsspwpw` is currently unused in set.mm, I suppose I'd think in terms of removing the whole thing (or moving it to a mathbox).

Reply:
Having an explicit upper bound on such a common construct as TopOn is important. Even if it's probably not that useful in CZF as you noted since powerset is not predicative, it is useful for instance to bound the set theoretic rank, or to tell that we stay within a given universe, or to bound the cardinality (finite or infinite).

Reply:
As for the theorem I proposed above, let's add it only once a need emerges.